api/hiveextension/v1beta1/agentclusterinstall_types.go

```diff
@@ -505,3 +512,20 @@ type ManifestsConfigMapReference struct {
 func init() {
 	SchemeBuilder.Register(&AgentClusterInstall{}, &AgentClusterInstallList{})
 }
+
+// MirrorRegistryConfigMapReference contains reference to a ConfigMap for mirror registry
+type MirrorRegistryConfigMapReference struct {
+	// Name is the name of the ConfigMap that this refers to
+	Name string `json:"name"`
+	// Namespace of the ConfigMap
+	Namespace string `json:"namespace"`
+}
+
+// MirrorRegistryConfiguration holds the given mirror registry configuration
+type MirrorRegistryConfiguration struct {
```

Comment:
Given that this is an internal representation of the mirror config (and that it's shared between infraenv and cluster) I think it should be somewhere internal rather than in the hiveextension API package.

Reply:
Moved to api/common